### PR TITLE
Fixed tests and added CI.

### PR DESCRIPTION
**Pull request by jezdez**
- Added Travis-CI
- Added coverage
- Added coveralls
- Fixed aggregator tests to not require PuSH server and not recreate objects again

### CODE REVIEW

**Comment by relekang** [comment]
Would it be better to list the test commands in tox.ini and run tox in make test? That way all tasks will run even if there is a flake8 error.

**Comment by jezdez** [comment]
Hm, I know what you're saying, I just use `make test` as a quick in-virtualenv test command instead of the much slower `tox` call during development. That would break that patterns :-/

**Comment by relekang** [comment]
What about having the command list in both files?

File: Makefile [hunk 1]
```diff
@@ -19,3 +19,8 @@ run:
 
 install:
 	pip install -r requirements/dev.txt
+
+test:
+	@flake8
+	@coverage run manage.py test aggregator contact docs fundraising legacy releases svntogit
+	@coverage report
```
Comment (timgraham) [inline Makefile]: Not sure how long it takes, but it seems a bit overkill for local development to generate a coverage report?

File: README.rst [hunk 2]
```diff
@@ -68,6 +74,36 @@ To run locally, do the usual:
...
+Our test results can be found here:
+
+    https://travis-ci.org/django/djangoproject.com
+
+For local development don't hesitate to install
+`tox <http://tox.readthedocs.org/>`_ to run the website's test suite.
+
+Then in the root directory (next to the ``manage.py`` file) run::
+
+    tox
```
Comment (timgraham) [inline README.rst]: I'd also like an explanation of what this does "under the hood", e.g. if you want to run the tests for an individual app, you can just use `manage.py test` as usual.

File: aggregator/tests.py [hunk 3]
```diff
@@ -14,38 +15,74 @@
 
 class AggregatorTests(TestCase):
 
-    def setUp(self):
+    @classmethod
+    @requests_mock.mock()
+    def setUpClass(cls, mocker):
```
Comment (timgraham) [inline aggregator/tests.py]: I think creating database objects in `setUpClass` isn't safe in 1.7. This is remedied in Django 1.8 with [TestCase.setUpTestData()](https://docs.djangoproject.com/en/dev/topics/testing/tools/#django.test.TestCase.setUpTestData).

File: setup.cfg [hunk 4]
```diff
@@ -1,3 +1,4 @@
 [flake8]
 exclude=djangodocs,migrations
 max-line-length=160
+ignore=W801,E128,E501,W402
```
Comment (timgraham) [inline setup.cfg]: I'm unsure why these need to be added. I don't see any errors on master currently.

File: aggregator/tests.py [hunk 5]
```diff
@@ -14,38 +15,74 @@
...
+                title="%s Item" % feed.title,
+                link=feed.public_url,
+                date_modified=datetime.datetime.now(),
+                guid=feed.title,
             )
 
     def test_feed_list_only_approved_and_active(self):
-        response = self.client.get(reverse('community-feed-list', kwargs={'feed_type_slug': self.feed_type.slug}))
+        response = self.client.get(
+            reverse('community-feed-list',
```
Comment (timgraham) [inline aggregator/tests.py]: This seemed fine on 1 line, but if the line was too long I think a more readable version would be:
```
url = reverse('community-feed-list', kwargs={...})
response = self.client.get(url)
```

File: aggregator/tests.py [hunk 6]
```diff
@@ -14,38 +15,74 @@
...
+            scm_url="/path/to/svn",
+            is_default=True,
+        )
 
         # Set up users who will get emailed
-        g = Group.objects.create(name=settings.FEED_APPROVERS_GROUP_NAME)
-        self.user = User.objects.create(username="Mr. Potato", email="mr@potato.com")
-        self.user.groups.add(g)
+        g, _ = Group.objects.get_or_create(
+            name=settings.FEED_APPROVERS_GROUP_NAME
```
Comment (timgraham) [inline aggregator/tests.py]: trailing comma

File: aggregator/tests.py (hunk 3 above)
Comment (jezdez) [inline aggregator/tests.py]: Bah, hm, I'd rather not wait for the yak shaving of jumping to 1.8 (and run an alpha), I'll make sure to update it to use `setUp` instead.

File: setup.cfg (hunk 4 above)
Comment (jezdez) [inline setup.cfg]: Just my common set of ignores ;) Take it as a precaution

File: aggregator/tests.py (hunk 5 above)
Comment (jezdez) [inline aggregator/tests.py]: wfm

File: README.rst (hunk 2 above)
Comment (jezdez) [inline README.rst]: Good idea, will do

File: Makefile (hunk 1 above)
Comment (jezdez) [inline Makefile]: *shrug* we can collect the data and document the possibility to run coverage report if that's better for you

**Pull request merged**